### PR TITLE
Refactor close account page to use Form

### PR DESCRIPTION
**Pull request by Puneet-here**
### Details

We are refactoring close account page to use Form
https://github.com/Expensify/App/issues/11341#issuecomment-1261440930
### Fixed Issues

$ https://github.com/Expensify/App/issues/11341
PROPOSAL: https://github.com/Expensify/App/issues/11341#issuecomment-1293263970

### Tests

1. Go to settings > Security > Close account
2. Try submitting the form without typing anything, verify you see error
3. Add correct contact method, verify the error goes away
4. Add wrong contact method and tap on screen to remove focus, verify you see error
5. Verify the button is red and is not disabled
6. Verify the button gets disabled when your are offline
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to settings > Security > Close account
2. Try submitting the form without typing anything, verify you see error
3. Add correct contact method, verify the error goes away
4. Add wrong contact method and tap on screen to remove focus, verify you see error
5. Verify the button is red and is not disabled
6. Verify the button gets disabled when your are offline
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/101883770/200542271-09a17fcf-1c9b-4d39-98ef-e6f1670e4aa9.mov

#### Mobile Web - Chrome

Screenshot 2022-11-08 at 4 35 53 PM

#### Mobile Web - Safari

Screenshot 2022-11-08 at 4 38 17 PM

#### Desktop

https://user-images.githubusercontent.com/101883770/200543847-da9d23a4-8fcb-4f2b-a54f-b57cbb10ccd1.mov

#### iOS

Screenshot 2022-11-08 at 4 38 37 PM

#### Android

Screenshot 2022-11-08 at 4 33 47 PM

### CODE REVIEW

File: src/pages/settings/Security/CloseAccountPage.js [hunk 1]
```diff
@@ -40,25 +35,33 @@ const propTypes = {
...
 class CloseAccountPage extends Component {
     constructor(props) {
         super(props);
 
-        this.state = {
-            reasonForLeaving: '',
-            phoneOrEmail: '',
-        };
+        this.onSubmit = this.onSubmit.bind(this);
+        this.validate = this.validate.bind(this);
```
Comment (thesahindia) [inline src/pages/settings/Security/CloseAccountPage.js:51]:
```suggestion
```
We don't have to bind this.

**Review by thesahindia — CHANGES_REQUESTED** [review]

File: src/languages/es.js [hunk 2]
```diff
@@ -399,16 +400,16 @@ export default {
...
         expensifyPassword: 'Contraseña de Expensify',
         error: {
-            invalidName: 'Por favor ingrese un nombre válido',
-            addressZipCode: 'Por favor ingrese un código postal válido',
-            debitCardNumber: 'Ingrese un número de tarjeta de débito válido',
+            invalidName: 'Por favor ingresa un nombre válido',
+            addressZipCode: 'Por favor ingresa un código postal válido',
+            debitCardNumber: 'Ingresa un número de tarjeta de débito válido',
             expirationDate: 'Por favor introduzca una fecha de vencimiento válida',
-            securityCode: 'Ingrese un código de seguridad válido',
```
Comment (Puneet-here) [inline src/languages/es.js:406]: This change is not related to this PR but added since it was asked [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1667921807574749?thread_ts=1667921293.250459&cid=C01GTK53T8Q)

**Comment by Beamanator** [comment]
@thesahindia 

> The other thing is that at mobile the modal covers the error above the button and it will be dismissed once user closes the modal
@Beamanator, what can we do here?

Ooh that's a very interesting point, it seems like we either need to do one of these:
1. Not show the error in the form (above the "Close account" button)
2. Not show the "Unable to close account" modal anymore

I call upon @shawnborton to assist with this question - @shawnborton on the "Close account" page we currently don't actually tell the user what error may have occurred when closing their account, we just show the "Unable to close account" modal (see [this comment](https://github.com/Expensify/App/pull/12534#issuecomment-1307526330)).

Is this a pattern we are ok with?
- If yes, I assume we want to avoid showing the exact error in the "Close account" form (and just show the modal)
- If not, maybe let's remove the modal & just show the error above the "Close account" button?

**Comment by thesahindia** [comment]
@shawnborton is OOO, should we tag the design team?

**Comment by Beamanator** [comment]
Aah shucks good point @thesahindia - currently @shawnborton is the whole @Expensify/design team 😅 

@thesahindia do you mind posting in #expensify-open-source to get some feedback on this idea? I think we can pause the PR-merge-timer while we wait for feedback

**Comment by shawnborton** [comment]
Just confirming that I like where we ended up in Slack, where we can omit the modal and just use the error message above the form button.

**Comment by Puneet-here** [comment]
@Beamanator @thesahindia I have updated the code to remove the modal, we need to remove the `<p>` tag from backend.

**Comment by Beamanator** [comment]
@Puneet-here thanks for the reminder, made a follow-up issue for myself here: https://github.com/Expensify/App/issues/12747

**Comment by Beamanator** [comment]
@thesahindia please feel free to test when you have time 👍

**Comment by thesahindia** [comment]
@Puneet-here, let's remove the modal part since we don't have a modal now
https://github.com/Expensify/App/blob/672dc9e0e16905d01bb2ce328fdeafbd15c89c7c/src/libs/actions/CloseAccount.js#L6

P.S. I have reviewed the code, it looks good. I will test it again in the morning. EOD for me 🌃

**Comment by thesahindia** [comment]
@Puneet-here, great work!
Looks good and works well 🎉

cc: @Beamanator 

## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2022-11-08 at 8 55 07 PM




</details>

<details>
<summary>Mobile Web - Chrome</summary>
Screenshot 2022-11-08 at 9 13 00 PM



</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2022-11-08 at 9 14 31 PM



</details>

<details>
<summary>Desktop</summary>
Screenshot 2022-11-08 at 8 23 36 PM



</details>

<details>
<summary>iOS</summary>
Screenshot 2022-11-08 at 9 13 51 PM



</details>

<details>
<summary>Android</summary>
Screenshot 2022-11-08 at 9 08 38 PM



</details>

**Review by thesahindia — APPROVED** [review]

**Comment by thesahindia** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2022-11-08 at 8 55 07 PM




</details>

<details>
<summary>Mobile Web - Chrome</summary>
Screenshot 2022-11-08 at 9 13 00 PM



</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2022-11-08 at 9 14 31 PM



</details>

<details>
<summary>Desktop</summary>
Screenshot 2022-11-08 at 8 23 36 PM



</details>

<details>
<summary>iOS</summary>
Screenshot 2022-11-08 at 9 13 51 PM



</details>

<details>
<summary>Android</summary>
Screenshot 2022-11-08 at 9 08 38 PM



</details>

**Review by thesahindia — DISMISSED** [review]

File: src/ONYXKEYS.js [hunk 3]
```diff
@@ -176,6 +173,7 @@ export default {
         REQUEST_CALL_FORM: 'requestCallForm',
         REIMBURSEMENT_ACCOUNT_FORM: 'reimbursementAccount',
         WORKSPACE_SETTINGS_FORM: 'workspaceSettingsForm',
+        CLOSE_ACCOUNT_FORM: 'closeAccount',
```
[the remark this answers is not in the document]
Comment (Beamanator) [reply src/ONYXKEYS.js:176]: But actually if we do this I'll also need to update the back-end.. Maybe we leave it as is for now

**Review by Beamanator — CHANGES_REQUESTED** [review]
A few small requested changes, happy to discuss :D

Also going to request @shawnborton check out the new Close Account page design, I don't know if I love the two error messages both showing in red, with red dots 🤔

File: src/pages/settings/Security/CloseAccountPage.js [hunk 4]
```diff
@@ -11,54 +12,47 @@ import compose from '../../../libs/compose';
...
-            reasonForLeaving: '',
-            phoneOrEmail: '',
-        };
+        this.onSubmit = this.onSubmit.bind(this);
+        CloseAccount.clearError();
+    }
+
+    componentWillUnmount() {
+        CloseAccount.clearError();
+    }
```
[the remark this answers is not in the document]
Comment (thesahindia) [reply src/pages/settings/Security/CloseAccountPage.js:43]: Right now we are following the same pattern on other pages. 

https://github.com/Expensify/App/blob/e81fab75896abc5f58ee443a3565fb4ad0bb839f/src/pages/settings/Payments/AddDebitCardPage.js#L56-L60

File: src/pages/settings/Security/CloseAccountPage.js [hunk 5]
```diff
@@ -71,21 +65,21 @@ class CloseAccountPage extends Component {
...
                 />
-                <ScrollView
-                    contentContainerStyle={[
-                        styles.flexGrow1,
-                        styles.flexColumn,
-                        styles.p5,
-                    ]}
+                <Form
+                    formID={ONYXKEYS.FORMS.CLOSE_ACCOUNT_FORM}
+                    validate={values => this.validate(values, userEmailOrPhone)}
```
[the remark this answers is not in the document]
Comment (thesahindia) [reply src/pages/settings/Security/CloseAccountPage.js:70]: We already have the variable above so we are using that, but I don't mind either way. 
https://github.com/Expensify/App/blob/e81fab75896abc5f58ee443a3565fb4ad0bb839f/src/pages/settings/Security/CloseAccountPage.js#L65

File: src/ONYXKEYS.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Beamanator) [reply src/ONYXKEYS.js:176]: Yeah I'm down with that, I'll resolve this comment

File: src/pages/settings/Security/CloseAccountPage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Beamanator) [reply src/pages/settings/Security/CloseAccountPage.js:43]: Aah interesting... I guess this makes sense - this form can only be submitted when we're online, so the response "should" be relatively quick, so if you navigate away you most likely saw the error so it should be fine to dismiss it

File: src/pages/settings/Security/CloseAccountPage.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (thesahindia) [reply src/pages/settings/Security/CloseAccountPage.js:70]: Let's change this @Puneet-here, don't forget to bind the validate method 😄

**Comment by shawnborton** [comment]
I agree that the red dot + red text is not my favorite, but I had posted some options in open source a while ago and this is what the group decided.

**Comment by Puneet-here** [comment]
Updated!
cc: @thesahindia @Beamanator

**Review by thesahindia — APPROVED** [review]

**Comment by thesahindia** [comment]
Thanks for the changes!

All yours @Beamanator

**Review by Beamanator — APPROVED** [review]
Sounds good, looks like we're good to go here!

File: src/components/Form.js [hunk 6]
```diff
@@ -48,6 +48,9 @@ const propTypes = {
     /** Should the button be enabled when offline */
     enabledWhenOffline: PropTypes.bool,
 
+    /** Whether the action is dangerous */
```
Comment (marcaaron) [inline src/components/Form.js:51]: I don't understand this - what does it mean for an action to be "dangerous"? 😕 

Why was this added to such a low level component..

Comment (Beamanator) [reply src/components/Form.js:51]: To me, "Dangerous" means if the user clicks the "Submit" button, the action being taken should only be taken with caution. I think it would be helpful if you can recommend a "better solution" or what you'd prefer seeing - I am guessing you are saying you'd like to see a more generic prop name, maybe something having to do with the submit button color?

Comment (marcaaron) [reply src/components/Form.js:51]: >To me, "Dangerous" means if the user clicks the "Submit" button, the action being taken should only be taken with caution.

>I am guessing you are saying you'd like to see a more generic prop name

No, I think it's _too_ generic. Which "action" is "dangerous"? It's a Form so the user can take many actions. It looks like this just controls the color of the button. So my suggestion would be something like `isSubmitButtonDangerous`, `isSubmitActionDangerous` etc.

Comment (Beamanator) [reply src/components/Form.js:51]: Aah ok I'm glad we got that clarified, I see what you mean that technically there can be many "actions" taken in a form, though I thought it was a bit self explanatory that the main "action" is submitting the form - but it's good to be clear 👍 I like your suggestions, I think `isSubmitActionDangerous` is a bit more clear to me - there should only be 1 "submit action" (clicking on the submit button) so I'll go with that - follow-up PR incoming!

Comment (Beamanator) [reply src/components/Form.js:51]: @marcaaron follow-up PR here: https://github.com/Expensify/App/pull/13981
